sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLViewSuite.scala

```diff
@@ -448,19 +476,105 @@ class SQLViewSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("Using view after change the origin view") {
+    withView("v1", "v2") {
+      sql("CREATE VIEW v1 AS SELECT id FROM jt")
+      sql("CREATE VIEW v2 AS SELECT * FROM v1")
+      withTable("jt2", "jt3") {
+        // Don't change the view schema
+        val df2 = (1 until 10).map(i => i + i).toDF("id")
+        df2.write.format("json").saveAsTable("jt2")
+        sql("ALTER VIEW v1 AS SELECT * FROM jt2")
+        // the view v2 should have the same output with the view v1
+        checkAnswer(sql("SELECT * FROM v2"), sql("SELECT * FROM v1"))
+        // the view v2 should have the same output with the table jt2
+        checkAnswer(sql("SELECT * FROM v2"), sql("SELECT * FROM jt2"))
+
+        // Change the view schema
+        val df3 = (1 until 10).map(i => i -> i).toDF("i", "j")
+        df3.write.format("json").saveAsTable("jt3")
+        sql("ALTER VIEW v1 AS SELECT * FROM jt3")
+        val e = intercept[AnalysisException] {
+          sql("SELECT * FROM v2")
+        }
+        assert(e.message.contains(
+          "The underlying schema doesn't match the original schema, expected " +
+            "STRUCT<`id`: INT> but got STRUCT<`i`: INT, `j`: INT>"))
+      }
+    }
+  }
+
+  test("Using view after drop the origin view") {
+    withView("v1", "v2") {
+      sql("CREATE VIEW v1 AS SELECT id FROM jt")
+      sql("CREATE VIEW v2 AS SELECT * FROM v1")
+      // Drop the referenced view
+      sql("DROP VIEW v1")
+      val e = intercept[RuntimeException] {
+        sql("SELECT * FROM v2")
+      }
+      assert(e.getMessage.contains(
+        "Failed to analyze the canonicalized SQL"))
+    }
+  }
```

Comment:
I suggest adding a variation of this test case, dropping the underlining table "jt" and v2 should be invalid. This will test the multiple-level expansion of a view. It is also no harm to check that v1 is also invalid in this new test case.